src/requests/sign-request.ts

```diff
@@ -113,11 +117,22 @@ export const signRequest = (
   const headers = canonicalRequest.headers ? normalizeHeaders(canonicalRequest.headers) : {}
   const body = canonicalRequest.body ?? ''
 
-  const { sortedHeaders, signedHeaders } = getSortedAndSignedHeaders(headers, timestamp)
+  const { sortedHeaders, signedHeaders } = getSortedAndSignedHeaders(
+    { ...headers, ...((contextHeaders as unknown) as Record<string, string>) },
```

Comment:
We would need to normalize context headers as we do for the rest to make sure signing works consistently. Maybe we can do as we do above so we don't need to force the type coercion here (if I am understanding the reason for `as` correctly)

Reply:
Key thing to keep in mind here is to guarantee contextHeaders values found in `headers` are overwritten. Basically merging two objects and we trust the second one more than the other